pkg/apis/events/v1alpha1/storage_types.go

```diff
@@ -91,6 +92,10 @@ type StorageSpec struct {
 	// Sink is a reference to an object that will resolve to a domain name to use
 	// as the sink.
 	Sink corev1.ObjectReference `json:"sink"`
+
+	// defaulterError is used exclusively to pass an error from the webhook defaulter to the webhook
+	// validator.
+	defaulterError error
```

Comment:
This is really gross, but I think it is the best way to do what I want. 

I could just do the defaulting in the reconciler, but then I need to be sure everything uses the local project variable, not spec.project. 

I could do the defaulting in the validator, but that feels wrong for some reason.

Reply:
Actually this method won't work at all, it seems we can't have unexported fields in a spec. Switched to defaulting in the webhook validator.

Reply:
This didn't work either, the changes validation made were ignored.